### PR TITLE
Start the ProcessManager in the unified shell

### PR DESCRIPTION
**Pull request by plasorak**
With this PR, one doesn't need to have 2 terminal windows open simultaneously to start the `drunc-unified-shell`, as the shell directly starts the process manager in a `multiprocess.Process`. To get this to work correctly, there is quite a bit of bookkeeping of the signal that needs to happen (not that on Linux the `multiprocess.Process` are forks):

1. I tied the Process Manager's fate to the drunc-unified-shell's main thread by using the code [here](https://gist.github.com/qxcv/fe5be4d14f855fedf7a5db723aad22c2). This uses magic incantations I don't fully understand, but it works correctly (and is close to a snippet at the top of `nanorc`'s SSHPM), and sends `SIGKILL` to the forked process when the main gets one.
2. When an exception gets thrown in the drunc-unified-shell's main, I SIGKILL the forked process, if it exists.
3. When the main gets a `SIGINT` (for example, `ctrl-c` from a user in the shell), this signal gets propagated to the forked process, so the fork needs to catch and ignore that signal in the case it runs with the unified shell.

That's all folks!

### CODE REVIEW

**Comment by plasorak** [comment]
With the last 2 commits, we can now specify
```json
    "command_address": "localhost:0",
```
In the Process manager json, so that we don't need to assign port ourselves for the process manager server.

File: src/drunc/unified_shell/shell.py [hunk 1]
```diff
@@ -3,22 +3,61 @@
...
+
+
+    from time import sleep
+    for _ in range(100):
+        if ready_event.is_set():
+            break
+        sleep(0.1)
+
+    if not ready_event.is_set():
+        raise DruncSetupException('Process manager did not start in time')
```
Comment (TiagoTAlves) [inline src/drunc/unified_shell/shell.py:49]: You raise a DruncSetupException but you don't import it and it gives a 
///bash
NameError: name 'DruncSetupException' is not defined
///

**Comment by TiagoTAlves** [comment]
Not sure I fully understand how this is meant to run so I attempted a couple of different things. 

I started with booting up a process manager `drunc-process-manager file://drunc/data/process-manager.json
`. then I tried to start up a unified-shell with `drunc-unified-shell grpc://localhost:{host_provided_by_pm}`, `drunc-unified-shell grpc://np04-srv-015:{host_provided_by_pm}` of which neither worked. Both gave me a 

```bash
DruncSetupException: Process manager did not start in time
```
once I added a `from drunc.exceptions import DruncSetupException`.

I then thought maybe  the unified shell would start the pm on its own and tried `drunc-unified-shell file://drunc/data/process-manager.json grpc://localhost:10054`. I might need some guidance on how its actually started.

**Comment by TiagoTAlves** [comment]
Once I added the the import statement I tested it and the unified shell worked

File: src/drunc/unified_shell/shell.py (hunk 1 above)
Comment (plasorak) [reply src/drunc/unified_shell/shell.py:50]: Thanks for catching this. Fixed

**Comment by TiagoTAlves** [comment]
Also works with K8s pm

**Review by TiagoTAlves — APPROVED** [review]

**Pull request merged**